ssl-config-core/src/main/resources/reference.conf

```diff
@@ -37,6 +38,18 @@ ssl-config {
   # otherwise the default will be used
   hostnameVerifierClass = null
 
+  sslParameters {
+    # translates to a setNeedClientAuth / setWantClientAuth calls
+    # "default" – leaves the (which for JDK8 means wantClientAuth and needClientAuth are set to false.)
+    # "none"    – `setNeedClientAuth(false)`
+    # "want"    – `setWantClientAuth(true)`
+    # "need"    – `setNeedClientAuth(true)`
+    clientAuth = "default"
+
+    # protocols (names)
+    protocols = []
+  }
```

Comment:
technically we could configure all SSLParameters here declaratively.
https://docs.oracle.com/javase/8/docs/api/javax/net/ssl/SSLParameters.html